### ISSUE TITLE
smartparens not working for ' and ", () throwing error, in rmarkdown polymode

### ISSUE THREAD

**Issue by MilesMcBain** [polymode/poly-R#2]
This had worked fine for me until I updated my polymode packages a few days ago.

In an .Rmd file, inside an R chunk, pressing " or ' just inserts a single character not a matched pair. A matched pair is inserted in the markdown buffer portions.

In plain .R files, using ESS, matched pairs for ' and " are inserted as expected.

Possibly related: inserting a parenthesis pair, () inside an Rmd R chunk generates an error: `Error running timer ‘sp-show--pair-function’: (wrong-type-argument characterp nil)`

Again I do not see this error when inserting parenthesis in the markdown buffer portions or when using plain .R files with ESS.

**Comment by vspinu** [comment]
I am using smartparens but I don't recall seeing such problems. Are you
sure smartparens mode gets activated in R portions in the first place?

Will have a look once I am back from vacation next week.

**Comment by vspinu** [comment]
I would need more info on this. The exact smartparens setup and an example of the file where the problem occurs. Could you please provide this info?

**Comment by MilesMcBain** [comment]
I use Spacemacs, with no custom `smartparens` config of my own. I believe the guts of it is here: https://github.com/syl20bnr/spacemacs/blob/master/layers/+spacemacs/spacemacs-editing/packages.el#L254

I have found some interesting further info re reproducing the problem. If you start a new blank .Rmd buffer and insert a single r chunk ie:

    ```{r}
    ```
And type `()` inside the chunk you will not see an error. However once you add a second chunk, you will see the error in the first chunk only and not the second, i.e.:

    ```{r}
    ## brackets typed in here cause error
    ## quotes typed in here do not have match inserted
    ```
    ```{r}
    ## brackets typed in here do not cause error
    ## quotes typed in here are matched.
    ```
A pattern appears with n chunks, where the errors appear in all chunks except the nth (last) one in the buffer.

FYI Polymode is not part of Spacemacs, I enabled it myself using this custom layer: https://github.com/MilesMcBain/spacemacs_cfg/blob/master/private/polymode/packages.el

**Comment by vspinu** [comment]
This is a bit complex. I have never used spacemacs or evil so it's a bit of high ramp for me. How do you add polymode layer, just put in .emacs.d? Also, is there a way to disable vim shortcuts evil layer in spacemacs?

BTW, if you have `toggle-debug-on-error` do you see a stacktrace?

**Comment by vspinu** [comment]
> Also, is there a way to disable vim shortcuts evil layer in spacemacs?

Nevermind about this one. It actually asks at installation startup.

**Comment by vspinu** [comment]
Ok, I spent a bit of time with spacemacs and I am afraid I don't see this on emacs27. 

This is what I do. I install spacemacs and set `dotspacemacs-editing-style' to 'emacs` in the .spaceemacs file. Then I add `poly-R` to `  `dotspacemacs-additional-packages` and add ` (require 'poly-R)` to the `dotspacemacs/user-init`. Then I open a tmp.rmd which I attach. Nothing of what you describe happens there. Smartparens mode is enabled in that buffer. 

I have discovered an issue with highlight-line mode but that's a different issue.

**Comment by MilesMcBain** [comment]
Emacs27! So you mean the master branch of the development version?

I am on 25.2.2, since I am a Debian/Ubuntu user and we're still waiting for 26 to land in the main repos. I'll try to see if I can reproduce using your steps on 25 and then 26.

**Comment by vspinu** [comment]
Ah, that's most likely it. There have been notable changes to parse-ppss in emacs 26. Your issue is almost surely related to those syntax caches. There are known issues on emacs 25 (some tests fail) but I don't have time to dig into those atm. Prefer to work for the future.

**Comment by MilesMcBain** [comment]
I can confirm the issue goes away with Emacs 26.1. I've created a PR to update the version note in the Polymode installation documentation.

**Comment by MilesMcBain** [comment]
I'm using the new version and it looks to be fixed on 25.2 :+1:

**Comment by vspinu** [comment]
Perfect. Thanks for letting me know.
